src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java

```diff
@@ -49,4 +83,52 @@ public void subscribeSuccess() {
         certSubscribeSuccess.incrementAndGet();
     }
 
+    /**
+     * Returns the Cert.SubscribeSuccess metric.
+     */
+    public long getSubscribeSuccess() {
+        return certSubscribeSuccess.get();
+    }
+
+    /**
+     * Increments the Cert.SubscribeError metric.
+     */
+    public void subscribeError() {
+        certSubscribeError.incrementAndGet();
+    }
```

Comment:
To elaborate more on my comment above, instead of having code that emits metrics call this directly, register this class as a listener for metric events and extract the metric from the event.